LayoutTests/fast/dom/HTMLLinkElement/link-stylesheet-load-once.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script>
+if (window.testRunner) {
+    testRunner.waitUntilDone();
+    testRunner.dumpAsText();
+}
+
+let loadCount = 0;
+function didLoad(element)
+{
+    if (!loadCount) {
+        document.querySelector('link').rel = 'StyleSheet';
```

Comment:
A few thoughts about how we might want to improve this test to help us further protect interoperability:

1) Make this part of WPT instead of a WebKit-only test.
2) Test whether the resources are reloaded, too, not just whether the load event fires.
3) Test other types of pairs of “same” rel attribute values, such as values that are literally identical, values that have case differences, values that have leading or trailing white space.
4) Test cases other than style sheets.

None of this seems strictly required, but honestly without more testing I am not sure that testing the **parsed** version of the link attribute for equality is precisely the right thing to do. Earlier on in my experience with the web platform I used to assume that in cases where something “changed” to the same effective value doing the minimum work was good for better performance and less flashiness. Later I learned that websites often rely on side effects of setting something to the same value, whether intentionally or not. Thus this is something that may need to be carefully tested and specified since intuition was not a good guide for me.

I’m not sure that all the browsers would agree about when the change to the attribute should trigger a reload, and any differences can lead to interoperability problems later.

Reply:
I think it's a bit tricky to detect how many loads had happened purely in JS. It probably needs to be a HTTP test. For now, I'm going to use `dumpResourceLoadCallbacks` to address (2). I'll add more cases for (3). I'd also a test for "preload" as well for (4) since that triggers load event as well.

Reply:
Tracking this issue in HTML spec as https://github.com/whatwg/html/issues/8138. Without that, there isn't a basis to add WPT.